# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/PRemoteWorkerDebuggerManager.ipdl
# Commit: e80b18ca9165
# Full Hash: e80b18ca916518c73d5435fe8f842d9f0e1b7585
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r cb743703a9ee -r e80b18ca9165 dom/workers/remoteworkers/PRemoteWorkerDebuggerManager.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/PRemoteWorkerDebuggerManager.ipdl	Tue Mar 25 09:13:20 2025 +0000
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include RemoteWorkerTypes;
+include protocol PRemoteWorkerDebugger;
+
+namespace mozilla {
+namespace dom{
+
+// PRemoteWorkerDebuggerManager is used to register a remote WorkerDebugger
+// (PRemoteWorkerDebuggerParent) for a Worker. It is a IPC between the Parent
+// process main thread and the Content process WorkerLauncher thread because
+// when construting a Worker, there is no directly IPC can be reused to directly
+// build up the IPC between the Parent process main thread and the Content
+// process Worker thread. However, the remote WorkerDebugger unregistration is
+// done by PRemoteWorkerDebugger itself, so PRemoteWorkerDebuggeManager has no
+// IPC method for unregister.
+
+[ChildProc=anydom]
+protocol PRemoteWorkerDebuggerManager
+{
+parent:
+  async Register(RemoteWorkerDebuggerInfo aDebuggerInfo,
+      Endpoint<PRemoteWorkerDebuggerParent> aParentEp);
+};
+
+} // namespace dom
+} // namespace mozilla